src/core.js

```diff
@@ -159,3 +149,8 @@ jQuery.fn.size = function() {
 	migrateWarn( "jQuery.fn.size() is deprecated; use the .length property" );
 	return this.length;
 };
+
+jQuery.parseJSON = function() {
+	migrateWarn( "jQuery.parseJSON is deprecated; use JSON.parse" );
```

Comment:
I see your point, for an ajax call, it's just that in this case the potential cause of the problem is on the server and outside the user's own code. That has been one of the most commonly reported issues since we changed it in 1.9. It should be possible for someone to resolve it on the client side via explicitly stating the `dataType` for example, but it would be better to give a warning about that than to throw a cryptic error on the console. Our cost of doing that is small here so I don't mind improving the way the problem is reported.

For `parseJSON` itself we definitely need the check to be restored so that it won't die on an empty string.